src/app/_services/api.service.ts

```diff
@@ -1,17 +1,33 @@
 import { Injectable } from '@angular/core';
 import { HttpClient } from '@angular/common/http';
 import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
 import { environment } from '../../environments/environment';
 import {
   BreakdownRequest,
+  BreakdownResult,
   BreakdownResults,
+  CountPercentageValue,
   GeneralResults,
-  IHash
+  GeneralResultsFormatted,
+  IHash,
+  IHashArray,
+  TargetCountryData,
+  TargetData,
+  TargetMetaData,
+  TargetMetaDataRaw
 } from '../_models';
 import { ISOCountryCodes } from '../_data';
+import { Cache } from '../_helpers';
 
 @Injectable({ providedIn: 'root' })
 export class APIService {
+  private readonly countries = new Cache(() => this.loadCountryData());
```

Comment:
Perhaps it is better to get a list of countries from the backend, not all countries have target data

Reply:
I think the back-end should supply empty results in that case, and the front end should handle the empty result.